noxfile.py

```diff
@@ -218,10 +206,7 @@ def run_examples(session):
 def run_scripts(session):
     """Run the scripts tests for Python scripts."""
     set_environment_variables(PYBAMM_ENV, session=session)
-    # Temporary fix for Python 3.12 CI. TODO: remove after
-    # https://bitbucket.org/pybtex-devs/pybtex/issues/169/replace-pkg_resources-with
-    # is fixed
-    session.install("setuptools", silent=False)
+    session.run("python", "-m", "pip", "install", "--upgrade", "pip")
```

Comment:
I mean, why not? It is not that essential and can be removed if you like, but upgrading pip to be up-to-date with bug and security patches is good, and also ensures compatibility (some newer packages may need the latest version of pip).

Reply:
Good practice does not typically mean that we have to control it. The version of pip in the environment is the responsibility of the user.

In something like CI or a docker image, then it is our responsibility to make sure we are using a secure version, but updating the pip version each time tests are run is probably not the correct approach.

Reply:
I see, I will the pip update commands

Reply:
> some newer packages may need the latest version of pip

This use case was valid, but it has largely diminished nowadays with the advent of the 2020 resolver and after pip 21.3+. Newer `pip` versions, even if they're a little outdated, can install almost all packages that play nicely with packaging tools and those that have precompiled wheels. It should be good to remove this command from all the sessions.

If you were wanting to do this to avoid the warning `pip` shows on the version being outdated, you can take a look at the `PIP_DISABLE_PIP_VERSION_CHECK` environment variable.